src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1701,6 +1702,12 @@ internal static void WidenFourAsciiBytesToUtf16AndWriteToBuffer(ref char outputB
                 Vector128<ulong> vecWide = Sse2.UnpackLow(vecNarrow, Vector128<byte>.Zero).AsUInt64();
                 Unsafe.WriteUnaligned<ulong>(ref Unsafe.As<char, byte>(ref outputBuffer), Sse2.X64.ConvertToUInt64(vecWide));
             }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                Vector128<byte> vecNarrow = AdvSimd.DuplicateToVector128(value).AsByte();
+                Vector128<ulong> vecWide = AdvSimd.Arm64.ZipLow(vecNarrow, Vector128<byte>.Zero).AsUInt64();
+                Unsafe.WriteUnaligned<ulong>(ref Unsafe.As<char, byte>(ref outputBuffer), vecWide.ToScalar());
```

Comment:
Does the use of `ToScalar` here produce acceptable codegen? In the SSE2 code paths, analyzing the x86 codegen revealed that `ConvertToUInt64` produced better codegen than `ToScalar`. I don't know if the arm64 JIT has a similar behavior.

Reply:
I believe we have fixed most of these cases and improved the codegen for things like `ToScalar`. If not, we should as that is preferable for readability.

Reply:
Yes, `ToScalar()` is optimized for arm64.